crowbar_framework/app/models/updater_service.rb

```diff
@@ -55,15 +63,23 @@ def create_proposal
 
   def apply_role_pre_chef_call(old_role, role, all_nodes)
     @logger.debug("Updater apply_role_post_chef_call: entering #{all_nodes.inspect}")
-    # Remove [:updater][:one_shot_run] flag from node
+    # Remove [:updater][...] flags from node
     all_nodes.each do |n|
       node = NodeObject.find_node_by_name n
       unless node[:updater].nil?
-        node[:updater][:one_shot_run] = false
-        @logger.debug("Updater apply_role_post_chef_call: delete [:updater][:one_shot_run] for #{node.name} (#{node[:updater].inspect}")
-        node.save
+        if node.role? "updater"
+          node[:updater][:one_shot_run] = false
+          @logger.debug("Updater apply_role_post_chef_call: delete [:updater][:one_shot_run] for #{node.name} (#{node[:updater].inspect}")
+          node.save
+        end
+        if node.role? "updater-crowbar-upgrade"
+          node[:updater][:upgrade_one_shot_run] = false
+          @logger.debug("Updater apply_role_post_chef_call: delete [:updater][:upgrade_one_shot_run] for #{node.name} (#{node[:updater].inspect}")
+          node.save
```

Comment:
I'd expect to have a call to ```node.set_state("crowbar-upgrade")``` here. That would cause the run_list to be updated before the chef-client run. This is related to https://github.com/crowbar/barclamp-updater/pull/13/files#r47887111